src/controllers/otpauth.controllers.ts

```diff
@@ -56,22 +64,29 @@ const verifyOtp = (req: Request, res: Response) => {
     return res.status(404).json({ message: "Invalid OTP" });
   }
   // this function will decode the token and get errors in case
-  const result = (err: Error, decoded: any) => {
+  const result = async (err: Error, decoded: any) => {
     if (err) {
       if (err.message.includes("expired")) {
         // When the token expires it shows that the token has expired
```

Comment:
This shoulfd be done through the middleware that verifies whether the token is valid on not. If it doesn't exist create one.